src/vmod_cookie.c

```diff
@@ -213,6 +220,58 @@ vmod_get(VRT_CTX, struct vmod_priv *priv, VCL_STRING name)
 }
 
 
+vre_t *
+compile_re(VRT_CTX, VCL_STRING expression) {
+	vre_t *vre;
+	const char *error;
+	int erroroffset;
+
+	vre = VRE_compile(expression, 0, &error, &erroroffset);
+	if (vre == NULL) {
+		VSLb(ctx->vsl, SLT_VCL_Log, "cookie: PCRE compile error at char %i: %s", erroroffset, error);
+		return(NULL);
+	}
+	return(vre);
+}
+
+VCL_STRING
+vmod_get_re(VRT_CTX, struct vmod_priv *priv, struct vmod_priv *priv_call, VCL_STRING expression)
+{
+	struct vmod_cookie *vcp = cobj_get(priv);
+	(void)ctx;
+	int i, ovector[VRE_MAX_GROUPS];
+	struct cookie *cookie = NULL;
+	struct cookie *current;
+	vre_t *vre = NULL;
+
+	if (expression == NULL || *expression == '\0')
+		return(NULL);
+
+	if (priv_call->priv == NULL) {
```

Comment:
I think this would need to happen under a lock, otherwise the two compilation calls will race and we'll leak

Reply:
An alternative would be to use the object interface and prep the RE in the object constructor